src/ngircd/login.c

```diff
@@ -201,9 +201,40 @@ Login_User_PostAuth(CLIENT *Client)
 	} else
 		IRC_SetPenalty(Client, 1);
 
+// do that autojoin stuff
+  Login_Autojoin(Client);
+
 	return CONNECTED;
 }
 
+GLOBAL void Login_Autojoin(CLIENT *Client)
+{
+  /** make an autojoin to each channel that is good for it **/
+  const struct Conf_Channel *conf_chan;
+  size_t i, n, channel_count = array_length(&Conf_Channels, sizeof(*conf_chan));
+  conf_chan = array_start(&Conf_Channels);
+  assert(channel_count == 0 || conf_chan != NULL);
+  REQUEST Req;
+
+  for (i = 0; i < channel_count; i++, conf_chan++) {
+    if(!conf_chan->autojoin) {
+      continue;
+    }
+    if (!conf_chan->name[0])
```

Comment:
This is not needed, Channel_IsValidName() should take care of the „no name given“ case.